_posts/2018-06-14-fio-tutorial.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Running FIO with pmem engines
+author: lukaszstolarczuk
+layout: post
+identifier: fio_tutorial
+---
+
+When we want to check performance of our library, either see if there was any regression or if our tweaks did the good job, we execute benchmarks. One of them is FIO. It helps us simulate synthethic traffic of reads and writes to pmem device. In this blog post we'll try to introduce this tool and explain how we use it.
+
+### Preparing environment
+For starters, links to needed software:
+* [FIO][1fioLink]
+* [NDCTL][ndctlLnk]
+And of course, to execute any FIO tests using pmem engines you need PMDK installed in your system.
+
+Tests can be run on [emulated pmem][13rde117] or real hardware (devices introduced as NVDIMM Firmware Interface Table (NFIT), according to ACPI Specification v6.0+). Either way you have to prepare device(s) before test execution.
+First step is to configure namespace (emulated pmem is delivered already as a namespace). For this step you can use command `ndctl create-namespace`. There are two main options to configure namespace in our case - either you choose mode `fsdax` or `devdax`.
+Namespace configured with **fsdax** is a block device with support of dax operations. It requires dax-capable filesystem (e.g. ext4 or xfs) and should be configured like:
+{% highlight bash %}
+ndctl create-namespace --type=pmem --mode=fsdax --region=X [--align=4k]
+mkfs.ext4 /dev/pmemX
+mkdir /mnt/pmem
+mount /dev/pmemX /mnt/pmem -o dax
+{% endhighlight %}
+In both cases (this and one below) region(s) should be created/maintained using administrative tool specific for your hardware. Optional parameter **align** can be used to change alignment, while default for both modes is 2M.
```

Comment:
I wouldn't refer to a later text.
"In all cases regions should be ..."

"for your platform".